provider/core/src/versions.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[derive(Debug, Clone, Default, Copy, PartialEq)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[non_exhaustive]
+/// Source data versions
+pub struct SourceVersions<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    /// The version of CLDR source data, if used during datagen.
+    pub cldr: Option<SourceVersion<'data>>,
```

Comment:
Observation/Issue: icu_provider has no concept of CLDR. I'd like to keep it agnostic if possible.

Reply:
personally I think it's important to have this but it's not a strong opinion